tests/integration/conftest.py

```diff
@@ -190,3 +193,65 @@ async def app(model: Model, app_no_runner: Application) -> AsyncIterator[Applica
     await assert_num_of_runners(unit, 1)
 
     yield app_no_runner
+
+
+@pytest_asyncio.fixture(scope="module")
+async def app_scheduled_events(
+    model: Model,
+    charm_path: Path,
+    app_name: str,
+    path: str,
+    token: str,
+    http_proxy: str,
+    https_proxy: str,
+    no_proxy: str,
+) -> AsyncIterator[Application]:
+    """Application with no token.
+
+    Test should ensure it returns with the application having one runner.
+
+    This fixture has to deploy an new application. The scheduled events are set
+    to one hour in other application to avoid conflicting with the tests.
+    Changes to the duration of scheduled interval only takes effect after the
+    next trigger. Therefore, it would take a hour for the duration change to
+    take effect.
+    """
+    subprocess.run(["sudo", "modprobe", "br_netfilter"])
+
+    await model.set_config(
+        {
+            "juju-http-proxy": http_proxy,
+            "juju-https-proxy": https_proxy,
+            "juju-no-proxy": no_proxy,
+            "logging-config": "<root>=INFO;unit=DEBUG",
+        }
+    )
+
+    application = await model.deploy(
+        charm_path,
+        application_name=app_name,
+        series="jammy",
+        config={
+            "path": path,
+            "token": token,
+            "virtual-machines": 0,
+            "denylist": "10.10.0.0/16",
+            "test-mode": "insecure",
+            "reconcile-interval": 2.0,
+            "update-interval": 2.0,
+        },
+    )
+    await model.wait_for_idle()
+
+    # Spawn one runner. Spawning one runner during deployment will cause timeout waiting for model.
+    unit = application.units[0]
+
+    await application.set_config({"virtual-machines": "1"})
+    action = await unit.run_action("reconcile-runners")
+    await action.wait()
+    await model.wait_for_idle(status=ACTIVE_STATUS_NAME)
+
+    # Wait until there is one runner.
+    await assert_num_of_runners(unit, 1)
```

Comment:
`assert_...` reads like a check compared to a wait function, should this be renamed?

Reply:
Renamed to `wait_till_num_of_runners`